src/SparseMatrix_DEN.cpp

```diff
@@ -0,0 +1,139 @@
+#include "SparseMatrix_DEN.hpp"
+#include <iostream>
+#include <memory>
+#include <algorithm>
+#include <string>
+#include <cmath>
+#include <cassert>
+#include <stddef.h>
+#include <vector>
+
+namespace SpMV
+{
+
+	////// Constructor Definition
+    template <class fp_type>
+    SparseMatrix_DEN<fp_type>::SparseMatrix_DEN(const size_t nrows, const size_t ncols) :
+        SparseMatrix<fp_type>::SparseMatrix(nrows, ncols)
+    {
+        std::cout << "Hello from SparseMatrix_DEN Constructor!" << std::endl;
+
+        // Initialize Matrix
+        _Matrix = std::make_unique<std::unique_ptr<fp_type[]>[]>(this->_nrows);
+
+        for (size_t i = 0; i < this->_nrows; ++i) {
+            _Matrix[i] = std::make_unique<fp_type[]>(this->_ncols);
+            std::fill(_Matrix[i].get(), _Matrix[i].get() + this->_ncols, fp_type(0)); // changed to fp_type(0)
+        }
+
+        // Set State to Initialized (Parent Constructor Already Does This)
+        this->_state = MatrixState::initialized;
+    }
+
+
+	////// Destructor Definition
+    template <class fp_type>
+    SparseMatrix_DEN<fp_type>::~SparseMatrix_DEN() {}
```

Comment:
Does the destructor automagically clear/deallocate/delete the `_Matrix` attribute?